scripts/build_llvm_project.py

```diff
@@ -0,0 +1,151 @@
+                        help='add custom argument to CMake')
+  buildopt.add_argument('-e', '--llvm-enable-projects', metavar='OPT', action='append', default=['clang'] if sys.platform == 'win32' else ['clang;openmp'],
+                        help='enable llvm projects to build (in quotation marks separated by semicolons e.g.: "clang;openmp")')
+  buildopt.add_argument('-c', '--use-ccache', action="store_true", default=False,
+                        help='Using ccache during the build (default: %(default)s)')
+  buildopt.add_argument('--cc', metavar='OPT', default=None,
+                        help='use specific C compiler')
+  buildopt.add_argument('--cxx', metavar='OPT', default=None,
+                        help='use specific C++ compiler')
+  buildopt.add_argument('-v', '--verbose', action='store_true', default=False,
+                        help='Verbose build (default: %(default)s')
+  arguments = parser.parse_args()
+  return arguments
+
+def generate_buildoptions(arguments):
+  install_root = Path(arguments.install_prefix)
+
+  base_cmake_args = [
+    f'-DCMAKE_INSTALL_PREFIX={install_root.as_posix()}',
+    f'-DCMAKE_BUILD_TYPE={arguments.build_type}',
+    f'-DCMAKE_TOOLCHAIN_FILE={arguments.toolchain}'
+  ]
+  if sys.platform == 'win32' and platform.uname()[4].lower() == 'arm64':
+    base_cmake_args.append('-GNMake Makefiles')
+  else:
+    generator = 'Ninja' if sys.platform == 'win32' else 'Unix Makefiles'
+    base_cmake_args.append(f'-G{generator}')
+
+  if arguments.use_ccache:
+    base_cmake_args.append('-DCMAKE_C_COMPILER_LAUNCHER=ccache')
+    base_cmake_args.append('-DCMAKE_CXX_COMPILER_LAUNCHER=ccache')
+
+  if arguments.cmake_param:
+    base_cmake_args.extend(arguments.cmake_param)
+
+  if arguments.c_compiler:
+    base_cmake_args.append(f'-DCMAKE_C_COMPILER={arguments.c_compiler}')
+
+  if arguments.cxx_compiler:
+    base_cmake_args.append(f'-DCMAKE_CXX_COMPILER={arguments.cxx_compiler}')
+
+  if arguments.verbose:
+    base_cmake_args.extend('-DCMAKE_VERBOSE_MAKEFILE=ON')
+
+  return base_cmake_args
+
+def normalize_builddir(project_srcdir, builddir, clean):
+  build_path = ''
+  if PurePath(builddir).is_absolute():
+    build_path = Path(builddir, PurePath(project_srcdir).name)
+  else:
+    build_path = Path(project_srcdir, builddir)
+
+  if clean and build_path.exists():
+    shutil.rmtree(build_path)
+
+  return build_path.as_posix()
+
+def configure_llvm(arguments):
+  build_path = normalize_builddir(
+    settings.LLVM_DIR, arguments.builddir, arguments.clean)
+
+  build_options = generate_buildoptions(arguments)
+  additional_options = [
+    f'-DLLVM_TARGETS_TO_BUILD={arguments.target}',
+    '-DLLVM_ENABLE_CLASSIC_FLANG=ON',
+  ]
+  if(arguments.llvm_enable_projects):
+    additional_options.append('-DLLVM_ENABLE_PROJECTS='+'; '.join(['%s']*len(arguments.llvm_enable_projects)) % tuple(arguments.llvm_enable_projects))
```

Comment:
Seems this expression is a bit complex, I think we can simplify it a bit.
What about this?
'-DLLVM_ENABLE_PROJECTS={}'.format(";".join([str(item) for item in arguments.llvm_enable_projects]))
Maybe you can leave the generator expression so just use ....format(";".join(arguments.llvm_enable_projects)))
Of course you should change the default value in the argument definition -> default=['clang', 'openmp']